packages/job-worker/src/ingest/expectedMediaItems.ts

```diff
@@ -220,7 +221,17 @@ export async function updateExpectedMediaItemForBucketAdLibAction(
 export function updateExpectedMediaItemsOnRundown(context: JobContext, cache: CacheForIngest): void {
 	const pieces = cache.Pieces.findFetch({})
 	const adlibs = cache.AdLibPieces.findFetch({})
-	const actions = cache.AdLibActions.findFetch({})
+	const actions: (AdLibAction | RundownBaselineAdLibAction)[] = cache.AdLibActions.findFetch({})
+
+	const baselinePiece = cache.RundownBaselineAdLibPieces.getIfLoaded()?.findFetch({})
```

Comment:
You don't want to be using `getIfLoaded()` here. 
If the collections aren't loaded, then when `saveIntoCache()` is called below, any ExpectedMediaItems for the baseline adlibs will be deleted.

Alternatively, you can use `getIfLoaded()` and supply a filter to `saveIntoCache()` so it leaves the baseline adlib items untouched. This is what is done in the ExpectedPackages flow

Reply:
So you propose that I use `isLoaded()` first and the use the `get()`?

Reply:
Looking at the implementation it looks like `saveInto` should be named something more along the lines of `wirteOver`.

Looking at the implementation in ExpectedPackages flow you utilize that you have `fromPieceType` in the ExpectedPackages collection, so you can identify where item is coming from and thus filter for it.

It doesn't look like we have something similar to that on the ExpectedMediaItems collections (and I think it might be overkill to implement that type since this is deprecated and we also want to move to Package Manager).

However, the `.get()` method seems to wait untill the collection is loaded and then return the data, so we could write something like this instead:
```
const baseLineActions = (await cache.RundownBaselineAdLibActions.get()).findFetch({})
actions.push(...baseLineActions)
```
That way we always have the data and `saveIntoCache` won't delete the mediaItems.

This would require changing the contract of `updateExpectedMediaItemsOnRundown` which is not ideal since it's deprecated mehtod, but the method is only used in one other place where it already calls a similar method that is async, so I don't think it will break anything:
```
export async function updateExpectedPackagesOnRundown(context: JobContext, cache: CacheForIngest): Promise<void> {
	// @todo: this call is for backwards compatibility and soon to be removed
	updateExpectedMediaItemsOnRundown(context, cache)
	await updateExpectedPlayoutItemsOnRundown(context, cache)
```

What are your thoughts on this approach @Julusian ?

Reply:
One thing to be aware of is that the `ExpectedMediaItems` were marked as deprecated something like 2 years ago when package manager was starting development. Which was way to early to do in my opinion, especially as NRK are still using this flow in r44. So those deprecated comments are really just noise/stale/junk that should be ignored.  
I don't think we have a plan on when we will actually deprecate it as bits `ExpectedMediaItems` and `ExpectedPlayoutItems` are rather interleaved, and package-manager can't replace `ExpectedPlayoutItems` yet.

I was going to write some stuff about performance of loading these collections unnecessarily, but it turns out I already made the same change for `ExpectedPlayoutItems` around 18 months ago https://github.com/nrkno/sofie-core/commit/209c36b4f2af48058640b801f0862be81ca7b2a2#diff-3cfbcc88a501bb55c33ede750614821d2c49c4c71bbf345cc24cb30fec2ed8a8

So yes load the collections like that (ideally in parallel using `Promise.all()`), like `ExpectedPlayoutItems` is doing https://github.com/tv2/tv-automation-server-core/blob/SOF-1242/ingestMediaBaselineAdlibWipes/packages/job-worker/src/ingest/expectedPlayoutItems.ts#L47-L81

Reply:
I have updated to the approach using `Promise.all()`